R/MsBackendDataFrame.R

```diff
@@ -145,8 +121,45 @@ setReplaceMethod("collisionEnergy", "MsBackendDataFrame", function(object, value
 })
 
 #' @rdname hidden_aliases
-setMethod("fromFile", "MsBackendDataFrame", function(object) {
-    .get_rle_column(object@spectraData, "fromFile")
+setMethod("dataOrigin", "MsBackendDataFrame", function(object) {
+    .get_rle_column(object@spectraData, "dataOrigin")
+})
+
+#' @rdname hidden_aliases
+setReplaceMethod("dataOrigin", "MsBackendDataFrame", function(object, value) {
+    if (!is.character(value) | length(value) != length(object))
+        stop("'value' has to be a 'character' of length ", length(object))
+    object@spectraData$dataOrigin <- .as_rle(as.character(value))
+    validObject(object)
+    object
+})
+
+#' @rdname hidden_aliases
+setMethod("dataOriginNames", "MsBackendDataFrame", function(object) {
+    if (is(object@spectraData$dataOrigin, "Rle"))
+        unique(object@spectraData$dataOrigin@values)
+    else unique(dataOrigin(object))
+})
+
+#' @rdname hidden_aliases
+setMethod("dataStorage", "MsBackendDataFrame", function(object) {
+    .get_rle_column(object@spectraData, "dataStorage")
+})
+
+#' @rdname hidden_aliases
+setReplaceMethod("dataStorage", "MsBackendDataFrame", function(object, value) {
+    if (!is.character(value) | length(value) != length(object))
+        stop("'value' has to be a 'character' of length ", length(object))
+    object@spectraData$dataStorage <- .as_rle(as.character(value))
+    validObject(object)
+    object
+})
+
+#' @rdname hidden_aliases
+setMethod("dataStorageNames", "MsBackendDataFrame", function(object) {
+    if (is(object@spectraData$dataStorage, "Rle"))
```

Comment:
We will use this function a lot actually. This will list all unique data storage files. The reason for this rather clumsy definition was the following: I did not want to call `unique(dataOrigin(object))` because this is first converting the `Rle` into a vector and then calling `unique` - rather inefficient. But you're right, calling `unique` on the `Rle` is as fast as `unique` on the `@values`. I will however still need an `if` statement, when the `dataStorage`/`dataOrigin` column is missing returning an `NA_character_` instead.

Reply:
Fixed now.

Reply:
Obsolete - method has gone away.